test/Table_Tests/src/Database/Postgres_Spec.enso

```diff
@@ -338,6 +341,96 @@ postgres_specific_spec connection db_name setup =
             w4.requested_type . should_equal (Value_Type.Decimal precision=Nothing scale=Nothing)
             w4.actual_type . should_equal Value_Type.Float
 
+        Test.specify "should round-trip timestamptz column, preserving intsant but converting to UTC" <|
+            table_name = Name_Generator.random_name "TimestampTZ"
+            table = connection.create_table table_name [Column_Description.Value "A" (Value_Type.Date_Time with_timezone=True)] primary_key=[]
+
+            dt1 = Date_Time.new 2022 05 04 15 30
+            dt2 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.utc)
+            dt3 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.parse "US/Hawaii")
+            dt4 = Date_Time.new 2022 05 04 15 30 zone=(Time_Zone.parse "Europe/Warsaw")
+            v = [dt1, dt2, dt3, dt4]
+
+            Problems.assume_no_problems <| table.update_rows (Table.new [["A", v]]) update_action=Update_Action.Insert
+
+            tz_zero = Time_Zone.parse "Z"
+            v_at_z = v.map dt-> dt.at_zone tz_zero
+            table.at "A" . to_vector . should_equal v_at_z
+            table.at "A" . to_vector . should_equal_tz_agnostic v
+
+            ## We also check how the timestamp column behaves with interpolations:
+               (See analogous test showing a less nice behaviour without timezones below.)
+            v.each my_dt-> Test.with_clue my_dt.to_text+" " <|
+                # Depending on test runner's timezone, the `my_dt` may be equal to 1 or 2 entries in `v`.
+                equals_tz_agnostic dt1 dt2 =
+                    dt1.at_zone tz_zero == dt2.at_zone tz_zero
+                local_equals = v.filter (equals_tz_agnostic my_dt)
```

Comment:
Here it looks like it's comparing all the elements of `v` to each other, ignoring time zones. Why would they ever be unequal?

Reply:
I will add a comment.

`equals_tz_agnostic` checks if the two date-time values represent the _same time instant_ regardless of timezone. It's not comparing the *local* date-time but the time instant.

E.g. `2023-10-20 16:26 UTC` and `2023-10-20 16:26 [Europe/Warsaw]` each have the same local date-time, but they represent different _time instants_ (differing by 2 hours). `2023-10-20 14:26 UTC` and `2023-10-20 16:26 [Europe/Warsaw]` would represent two same time instants and would be deemed `equals_tz_agnostic`.

I will rename to `is_same_time_instant`.